hibernate-core/src/test/java/org/hibernate/test/hql/HQLTest.java

```diff
@@ -1538,4 +1501,11 @@ public void testComponentNoAlias() throws Exception {
 		compileWithAstQueryTranslator( "from Human where name.first = 'Gavin'", false);
 	}
 
+	@Test
+	public void testCountDistinctOfSubquery(){
+		createNewQueryTranslator(
+				"select count (DISTINCT (SELECT function('lower', 'DAY') from Bar)) from Bar"
```

Comment:
This is not a valid SELECT COUNT expression. Let's take [PostgreSQL](https://www.postgresql.org/docs/current/static/functions-aggregate.html) for instance. The COUNT accepts a list of columns, but I've never seen it used with a subquery.

On what DB did you test this query?